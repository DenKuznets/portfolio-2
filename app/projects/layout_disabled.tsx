
import { ReactNode } from 'react';

const ProjectsLayout = ({ children }: { children: ReactNode }) => {
    return (
        <>
            layout
            {children}
        </>
    );
};

export default ProjectsLayout;
