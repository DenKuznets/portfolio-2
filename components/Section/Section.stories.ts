import type { Meta, StoryObj } from '@storybook/react';
import Section from './Section';

const meta = {
    title: 'Section',
    component: Section,
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
    parameters: {
        // More on how to position stories at: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'fullscreen'
    }
} satisfies Meta<typeof Section>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        projectId: 1
    }
};
