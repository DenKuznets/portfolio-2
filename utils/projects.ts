export interface Project {
    id: number;
    name: string;
    tech: string[];
    img: string;
    github: string;
    demo: string;
    description: string;
}

export const getProject = (id: number) =>
    projects.find((proj) => proj.id === id);

export interface techListType {
    react: string,
    nextjs: string,
    tailwind: string,
    ts: string,
    mui: string,
    redux: string,
    rest: string,
    unittest: string,
    storybook: string,
    sc: string,
    classic: string,
}

export const techList: techListType = {
    react: 'REACT',
    nextjs: 'NEXTJS',
    tailwind: 'TAILWIND',
    ts: 'TS',
    mui: 'MUI',
    redux: 'REDUX',
    rest: 'REST',
    unittest:"UNIT TESTING",
    storybook: 'STORYBOOK',
    sc: 'STYLED-COMPONENTS',
    classic:"HTML CSS JS"
};

export const projects: Project[] = [
    {
        id: 1,
        name: 'rock paper scissors',
        tech: [
            techList.react,
            techList.mui,
            techList.ts,
            techList.storybook,
            techList.unittest,
        ],
        img: 'rpc.webp',
        github: 'https://github.com/DenKuznets/rock-paper-scissors',
        demo: 'https://rock-paper-scissors-kappa-tawny.vercel.app/',
        description: `Игра "Камень Ножницы Бумага" по дизайну https://www.frontendmentor.io/challenges/rock-paper-scissors-game-pTgwgvgH`
    },
    {
        id: 2,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'tes.webp',
        github: 'https://github.com/DenKuznets/tes.loc',
        demo: 'https://tul-energo-service.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 3,
        name: 'тестовое задание',
        tech: [techList.react, techList.sc, techList.rest],
        img: 'test1.webp',
        github: 'https://github.com/DenKuznets/dls-test',
        demo: 'https://dls-test.vercel.app/',
        description: `Тестовое задание по макету Figma и ТЗ: 
                        1) Горизонтальное меню - sticky, исчезает после 200px прокрутки
                        2) Мобильное меню - плавное скрытие/открытие
                        3) Подменю - только CSS
                        4) Получение данных fetch()/axios
                        5) Отображение списка постов - flex
                        6) Поиск постов - по заголовкам и тексту
                        7) Полная информация о посте по клику на пост - modal window     
                        `
    },
    {
        id: 4,
        name: 'тестовое задание для web-студии',
        tech: [techList.react, techList.sc],
        img: 'webstudio.webp',
        github: 'https://github.com/DenKuznets/testovoe-alexmedia',
        demo: 'https://denkuznets.github.io/testovoe-alexmedia/',
        description: `Тестовое задание по макету Figma и ТЗ: 
                        1) Сделать липкое меню.
                        2) Мобильное и декстпное меню должно в html быть одним блоком. 
                        3) Добавить Анимации по своему усмотрению. Можно использовать сторонние библиотеки , но без jquery.(не обязательно)
                        4) Cделать popup и в форме предусмотреть валидацию, Пока не заполненны все поля кнопка должна быть неактивна.(не обязательно)
                        5) Ховеры на ссылки, по своему усмотрению.
                        6) Мобильная верстка`
    },

    {
        id: 5,
        name: 'портфолио',
        tech: [techList.react, techList.sc, techList.ts],
        img: 'portfolio.webp',
        github: 'https://github.com/DenKuznets/masha-portfolio',
        demo: 'http://davinchushka.vercel.app/',
        description: `шаблон портфолио по дизайну figma`
    },
    {
        id: 6,
        name: 'квиз',
        tech: [techList.react, techList.rest, techList.sc],
        img: 'quizz.webp',
        github: 'https://github.com/DenKuznets/quizzical-app',
        demo: 'https://denkuznets.github.io/quizzical-app/',
        description: `Пример использования API. По макету figma в рамках задания сайта scrimba.com, курс Learn React`
    },
    {
        id: 7,
        name: 'компонент диаграмма расходов',
        tech: [techList.react, techList.ts, techList.sc],
        img: 'component.webp',
        github: 'https://github.com/DenKuznets/Expenses-chart-component',
        demo: 'https://denkuznets.github.io/Expenses-chart-component/',
        description: `Основная сложность состояла в том, что бы придумать как реализовать отображение диаграммы. Кроме того, пробовал свои силы в написании компонентов react с использованием typescript. Задание с сайта frontendmentor.io`
    },
    {
        id: 8,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'tehavto.webp',
        github: 'https://github.com/DenKuznets/teh-avto.loc',
        demo: 'https://avtodom-techauto.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 9,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'bmk.webp',
        github: 'https://github.com/DenKuznets/bmk-new.loc',
        demo: 'https://bmk78.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 10,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'cleaning.webp',
        github: 'https://github.com/DenKuznets/cleaning-chistulya.loc',
        demo: 'https://chistylya.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 11,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'myp.webp',
        github: 'https://github.com/DenKuznets/myp-comfort',
        demo: 'https://mup-komfort.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 12,
        name: 'мое портфолио',
        tech: [techList.nextjs, techList.ts, techList.tailwind, techList.storybook, techList.unittest],
        img: 'myportfolio.webp',
        github: 'https://github.com/DenKuznets/portfolio',
        demo: 'https://denkuznets.vercel.app/',
        description: `Этот сайт портфолио я разработал сам, вдохновляясь примерами таких работ как https://stefantopalovicdev.vercel.app/ , https://caferati.me/portfolio , https://brittanychiang.com/ и http://riccardozanutta.com/ . 
                    За основу дизайна использовал этот шаблон https://themeforest.net/item/orido-personal-portfolio-website-figma-template/screenshots/37527613?index=1`
    },

    {
        id: 13,
        name: "drag'n'drop на react",
        tech: [techList.react, techList.redux],
        img: 'dnd.webp',
        github: 'https://github.com/DenKuznets/react-dnd',
        demo: 'https://react-dnd-ten.vercel.app/',
        description: `В рамках тестового задания реализован react dnd. CSS внимание не уделялось, поскольку работа велась в условиях сильного ограничения по времени задания.  
                    ТЗ: 
                    Реализовать простой 2д планировщик.
                        1) Сделать блок со списком объектов на выбор(это могут быть 2д
                            фотографии столов, стульев, перегородок).
                        2) Сделать доску на которое можно добавлять эти объекты и
                            перемещать их мышкой. 
                        3) Должна быть возможность сохранения расстановки в
                                файл(сохранять id объекта и его координаты).
                        4) Также должна быть возможность импорта этого файла с
                                координатами и отображения сохраненной расстановки.
                        5) Ограничения на стек технологий:JavaScript React Redux Остальное на ваше усмотрение`
    }
];
