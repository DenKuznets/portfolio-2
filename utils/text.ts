const text = {
    showmore: 'Подробнее',
    backButton: 'Назад',
    myWorks: 'Мои проекты',
    allWorks: 'Все проекты',
    examples: 'Примеры работ',
    aboutMe: 'Обо мне',
    links: {
        inst: 'https://www.instagram.com/denis_kyznecov/',
        git: 'https://github.com/DenKuznets',
        tg: 'https://t.me/DenKuznets',
        project: '/projects/'
    },
    hero: {
        fname: 'Денис',
        lname: 'Кузнецов',
        profession: 'front-end разработчик',
        aboutme: 'верстаю сайты на react и styled-components',
        stack: 'СТЭК: react, nextjs, ts, mui, tailwind, styled-components, redux, rest, graphql, jest, react-testing-library, storybook',
        contactme: 'Связаться',
        socials: 'я в соц сетях'
    }
};

export default text;
