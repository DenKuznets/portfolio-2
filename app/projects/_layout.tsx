import { techList } from '@/utils/projects';
import Link from 'next/link';
import { ReactNode } from 'react';

const ProjectsPage = ({ params, children }: { params: { filter: string }, children: ReactNode }) => {
    console.log('layout',params.filter);
    
    return (
        <div className="pt-8">
            <div className="flex flex-wrap gap-x-4 gap-y-2">
                {Object.keys(techList).map((key) => (
                    <Link
                        className={`border p-1 cursor-pointer ${params.filter === techList[key] && 'bg-red-500'}`}
                        key={techList[key]}
                        href={`/projects/${key}`}
                    >
                        {techList[key]}
                    </Link>
                ))}
            </div>
            {children}
        </div>
    );
};

export default ProjectsPage;
