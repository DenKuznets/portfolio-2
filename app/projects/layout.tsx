import { techList } from '@/utils/projects';
import Link from 'next/link';
import { ReactNode } from 'react';

const ProjectsPage = ({ params, children }: { params: { filter: string }, children: ReactNode }) => {
    console.log(params.filter);
    
    return (
        <div className="pt-8">
            <div className="flex flex-wrap gap-x-4 gap-y-2">
                {Object.keys(techList).map((key) => (
                    <Link
                        className={`border p-1 cursor-pointer`}
                        key={techList[key]}
                        href={`/projects/${key}`}
                    >
                        {techList[key]}
                    </Link>
                ))}
            </div>
            {children}
        </div>
    );
};

export default ProjectsPage;
