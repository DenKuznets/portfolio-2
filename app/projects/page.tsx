import ProjectCard from '@/components/ProjectCard/ProjectCard';
import { projects as projectsList, techListType } from '@/utils/projects';
import Link from 'next/link';
import { techList } from '@/utils/projects';
import { SearchParams } from '@/types';
import text from '@/utils/text';

type Props = {
    searchParams: SearchParams;
};

const page = ({ searchParams }: Props) => {
    return (
        <div className="container px-2 md:px-4">
            <div className="mx-auto flex  max-w-3xl flex-wrap justify-center gap-x-4 gap-y-2 sm:pb-6">
                <Link
                    className={`cursor-pointer rounded border p-1 text-xs uppercase no-underline  sm:p-2 sm:text-sm md:text-base ${
                        !searchParams.filter && 'bg-customYellow '
                    }`}
                    href={text.links.project}
                >
                    {text.all}
                </Link>
                {Object.keys(techList).map((key) => (
                    <Link
                        className={`cursor-pointer rounded border p-1 text-xs sm:p-2 sm:text-sm  md:text-base ${
                            searchParams.filter === key &&
                            'bg-customYellow'
                        }`}
                        key={techList[key as keyof techListType]}
                        href={`/projects?filter=${key}`}
                    >
                        {techList[key as keyof techListType]}
                    </Link>
                ))}
            </div>

            <div className="max-w-[1440px] mx-auto flex flex-col flex-wrap gap-4 pt-4 sm:grid sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
                {projectsList
                    .filter((project) =>
                        searchParams.filter
                            ? project.tech.includes(
                                  techList[
                                      searchParams.filter as keyof techListType
                                  ]
                              )
                            : true
                    )
                    .map((project) => (
                        <ProjectCard key={project.id} projectId={project.id} />
                    ))}
            </div>
        </div>
    );
};

export default page;
