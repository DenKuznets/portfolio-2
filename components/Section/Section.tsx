import React from 'react';
import { getProject } from '../../utils/projects';
import Link from 'next/link';
import MyLink from '../MyLink/MyLink';
import text from '../../utils/text';
import Image from 'next/image';

export interface SectionProps extends React.HTMLAttributes<HTMLDivElement> {
    projectId: number;
}

const Section = ({className, projectId }: SectionProps) => {
    const project = getProject(projectId);
    return project ? (
        <div className={`container lg:flex lg:gap-8 justify-between ${className ? className : ""}`}>
            <div>
                <h3 className="font-bold leading-normal mb-4">
                    {project.name}
                </h3>
                <ul className="flex flex-wrap gap-3 items-center mb-4">
                    {project.tech.map((obj) => (
                        <li
                            className="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-black border"
                            key={obj}
                        >
                            {obj}
                        </li>
                    ))}
                </ul>
                <Link href={project.github}>Github</Link> |{' '}
                <Link href={project.demo}>Demo</Link>
                <MyLink href={`/allworks/${projectId}`} className="mt-4">
                    {text.showmore}
                </MyLink>
            </div>

            <Link
                className="relative hidden md:block mt-8 shadow-2xl overflow-hidden lg:mt-0"
                href={`/allworks/${projectId}`}
            >
                <Image
                    className='filter-none hover:saturate-200 transition-all duration-300'
                    width={800}
                    height={600}
                    src={`/images/works-preview/${project.img}`}
                    alt={`${project.name} project image`}
                />
            </Link>
        </div>
    ) : (
        <div>project not found</div>
    );
};

export default Section;
