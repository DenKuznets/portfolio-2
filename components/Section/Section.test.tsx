import { render, screen } from '@testing-library/react';
import Section from './Section';
import '@testing-library/jest-dom';
import {  projects } from '@/utils/projects';
import text from '../../utils/text';

const project = projects[0];

test('renders correctly', () => {
    render(<Section projectId={project.id} />);
    const header = screen.getByRole('heading', { level: 3 });
    const listItems = screen.getAllByRole('listitem');
    const gitLink = screen.getByRole('link', { name: 'Github' });
    const demoLink = screen.getByRole('link', { name: 'Demo' });
    const showmoreLink = screen.getByRole('link', { name: text.showmore });
    const img = screen.getByAltText(`${project.name} project image`);

    expect(header).toHaveTextContent(project?.name);
    expect(listItems).toHaveLength(project.tech.length);
    expect(gitLink).toHaveAttribute('href', project.github);
    expect(demoLink).toHaveAttribute('href', project.demo);
    expect(showmoreLink).toHaveAttribute('href', `/allworks/${project.id}`);
    expect(img).toBeInTheDocument();
});

test('renders Section unchanged', () => {
    const { container } = render(<Section projectId={1} />);
    expect(container).toMatchSnapshot();
});
