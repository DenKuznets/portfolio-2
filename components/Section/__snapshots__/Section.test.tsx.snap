// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Section unchanged 1`] = `
<div>
  <div
    class="container lg:grid grid-cols-2 items-center lg:gap-8 py-8 justify-between "
    data-testid="section"
  >
    <div>
      <h3
        class="font-bold leading-normal mb-4 first-letter:capitalize"
      >
        rock paper scissors
      </h3>
      <ul
        class="flex flex-wrap gap-3 items-center mb-4"
      >
        <li
          class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
        >
          react
        </li>
        <li
          class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
        >
          mui
        </li>
        <li
          class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
        >
          ts
        </li>
        <li
          class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
        >
          unit testing
        </li>
      </ul>
      <a
        class="text-sm sm:text-lg"
        href="https://github.com/DenKuznets/rock-paper-scissors"
      >
        Github
      </a>
       
      |
       
      <a
        class="text-sm sm:text-lg"
        href="https://rock-paper-scissors-kappa-tawny.vercel.app/"
      >
        Demo
      </a>
      <a
        class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            mt-4 mb-4 lg:mb-0 hover:outline-inherit"
        href="/allworks/1"
      >
        Подробнее
        <svg
          class="fill-current"
          height="30px"
          style="stroke: currentColor;"
          viewBox="0 0 24 24"
          width="30px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 17L17 7M17 7H8M17 7V16"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
    </div>
    <a
      class="relative mt-8 lg:mt-0"
      href="/allworks/1"
    >
      <img
        alt="rock paper scissors project image"
        class="filter-none shadow-2xl hover:saturate-150 transition-all duration-300"
        data-nimg="1"
        decoding="async"
        height="600"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1920&q=75"
        srcset="/_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=828&q=75 1x, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1920&q=75 2x"
        style="color: transparent;"
        width="800"
      />
    </a>
  </div>
</div>
`;
